edb/edgeql/compiler/config.py

```diff
@@ -291,7 +304,41 @@ def _validate_global_op(
                        requires_restart=False,
                        backend_setting=None,
                        is_system_config=False,
-                       affects_compilation=False)
+                       affects_compilation=False,
+                       ptr=None)
+
+
+def _enforce_pointer_constraints(
+        ptr: s_pointers.Pointer, expr: irast.Set, *,
+        ctx: context.ContextLevel,
+        for_obj: bool) -> None:
+    constraints = ptr.get_constraints(ctx.env.schema)
+    for constraint in constraints.objects(ctx.env.schema):
+        if constraint.issubclass(
+            ctx.env.schema,
+            ctx.env.schema.get('std::exclusive', type=s_constr.Constraint),
+        ):
+            continue
+
+        with ctx.detached() as sctx:
+            sctx.partial_path_prefix = expr
+            sctx.anchors = ctx.anchors.copy()
+            sctx.anchors[qlast.Subject().name] = expr
+
+            final_expr = constraint.get_finalexpr(ctx.env.schema)
+            assert final_expr is not None and final_expr.qlast is not None
+            ir = dispatch.compile(final_expr.qlast, ctx=sctx)
+
+        result = ireval.evaluate(ir, schema=ctx.env.schema)
```

Comment:
Ideally we need to check if the constraint you added can be staevaled at the point of config type definition.